archivessnake/edit_notes.py

```diff
@@ -7,61 +7,59 @@
 import time
 
 from asnake.aspace import ASpace
+from asnake.utils import walk_tree
 #from fuzzywuzzy import fuzz
 from rapidfuzz import fuzz
 
-AS_REPOSITORY = 2
 OUTPUT_FILENAME = "out.csv"
 NOTE_TYPE_CHOICES = ["bioghist", "accessrestrict", "odd", "abstract", "arrangement", "userestrict", "fileplan", "acqinfo", "langmaterial", "physdesc", "prefercite", "processinfo", "relatedmaterial", "separatedmaterial"]
 ACTION_CHOICES = ["modify", "delete"]
 LEVEL = ["collection", "file", "series", "item", "all"]
 CONFIDENCE_RATIO = 97
-spreadsheet_path = os.path.join(os.path.abspath(os.path.dirname(__file__)), OUTPUT_FILENAME)
 
-def process_tree(args, resource):
+def process_tree(args, resource_id):
     """Iterates through a given collection, file, or series provided by user input. Finds note content that matches user input and then deletes or modifies relevant notes according to user preference.  """
-    for record in resource.tree.walk:
+    for record in walk_tree(resource_id, client):
         updated = False
-        aojson = record.json()
-        if args.level in [record.level, "all"]:
-            notes = aojson.get("notes")
+        if args.level in [record['level'], "all"]:
+            notes = client.get(record['notes']).json()
             for idx, note in reversed(list(enumerate(notes))):
-                if note["type"] == args.note_type:
-                    for subnote in note.get("subnotes"):
-                        content = subnote["content"]
+                if note['type'] == args.note_type:
+                    for subnote in note.get('subnotes'):
+                        content = subnote['content']
                         if contains_match(content, args.search_string):
                             if args.action == "delete":
                                 del notes[idx]
                                 print("{} note deleted from object {}".format(
-                                    args.note_type, record.uri))
+                                    args.note_type, record['uri']))
                             if args.action == "modify":
                                 print("{} note was originally {} and was changed to {} in object {}".format(
-                                    args.note_type, content, args.replace_string, record.uri))
-                                subnote["content"] = args.replace_string
+                                    args.note_type, content, args.replace_string, record['uri']))
+                                subnote['content'] = args.replace_string
```

Comment:
I would consider breaking this out into its own method, something like `handle_matching_note`?